rfc822/parser.go

```diff
@@ -57,30 +57,33 @@ func (section *Section) Literal() []byte {
 	return section.literal[section.header:section.end]
 }
 
-func (section *Section) Children() []*Section {
+func (section *Section) Children() ([]*Section, error) {
 	if len(section.children) == 0 {
 		if err := section.load(); err != nil {
-			panic(err)
+			return nil, err
 		}
 	}
 
-	return section.children
+	return section.children, nil
 }
 
-func (section *Section) Part(identifier ...int) *Section {
+func (section *Section) Part(identifier ...int) (*Section, error) {
 	if len(identifier) > 0 {
-		children := section.Children()
+		children, err := section.Children()
+		if err != nil {
+			return nil, err
+		}
 
 		if identifier[0] <= 0 || identifier[0]-1 > len(children) {
-			return nil
+			return nil, nil
```

Comment:
return error here? because the requested part doesn't exist

Reply:
Good point. fixed.